src/main/java/org/robotframework/swing/table/CellValueExtractor.java

```diff
@@ -13,23 +13,45 @@
 import org.robotframework.swing.common.SmoothInvoker;
 
 public class CellValueExtractor {
+    public enum TextSource {AUTO, MODEL}
+
     private JTableOperator jTableOperator;
 
     public CellValueExtractor(JTableOperator jTableOperator) {
         this.jTableOperator = jTableOperator;
     }
 
+    public String textOf(int row, int col, TextSource source) {
+        if (source == TextSource.MODEL)
+            return getTextFromTableModel(row, col);
+        else
+            return getTextWithDefaultStrategy(row, col);
+    }
+
     public String textOf(int row, int col) {
+        return textOf(row, col, TextSource.AUTO);
+    }
+
+    public String getTextWithDefaultStrategy(int row, int col) {
         try {
-            Component cellRendererComponent = getCellRendererComponent(row, col);
-            if (isButtonBasedRenderer(cellRendererComponent))
-                return new Boolean(((AbstractButton) cellRendererComponent).isSelected()).toString();
-            return coerceToWithText(cellRendererComponent).getText();
+            return getTextFromCellComponent(row, col);
         } catch (AllMethodsNotImplementedException e) {
-            return wrapElementToWithText(row, col).getText();
+            return getTextFromTableModel(row, col);
         }
     }
 
+    private String getTextFromTableModel(int row, int col) {
+        return wrapElementToWithText(row, col).getText();
+    }
+
+    private String getTextFromCellComponent(int row, int col) {
+        Component cellRendererComponent = getCellRendererComponent(row, col);
+        System.out.println(cellRendererComponent.getClass().getName());
```

Comment:
I dont mind adding debug logging, but some context should be provided "Getting text from element of type xxxx" and logged perhaps at debug level?